WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -248,17 +248,25 @@ def update_events
     end
   end
 
+  def get_nearby_competitions(competition)
+    nearby_competitions = competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10]
+    nearby_competitions.select!(&:isConfirmed?) unless current_user.can_view_hidden_competitions?
+    nearby_competitions || []
```

Comment:
`select!` returns `nil` for an empty array (but the array remains empty), so in this case as it's the last statement it's possible that `nil` appears.
As it's the last line you can use the `select` instead, without any additional assignment (just remove `!`).